### ISSUE TITLE
Multi select actions

### ISSUE THREAD

**Issue by contrapunctus-1** [MarcusWolschon/osmeditor4android#881]
It would be nice to have the following actions when selecting multiple objects -

- [x] 1. Copy - some features, e.g. pedestrian subways and bridges, are often very similar and to be able to copy them would save a lot of time.
- [x] 2. Add to relation - not so important, as it is also possible to add members to a relation after having selected the relation. It might be nice to be able to select multiple objects and add them to multiple relations, which this might enable. (Select objects -> add to relation -> objects remain selected, so you can repeat step 2 as many times as you want.)

**Comment by contrapunctus-1** [comment]
Also - "align in circle" (presumably after #828 )

**Comment by simonpoole** [comment]
1. added in https://github.com/MarcusWolschon/osmeditor4android/commit/037abc301be7f6c22a6561b4023e19f58c5d691a

**Comment by simonpoole** [comment]
Bump to 14.1

**Comment by jajajaneeneenee** [comment]
Not sure if this this issue is the right place, because it concerns multi-selected nodes (not objects) – but one of my most wanted features in Vespucci is (functionality like in JOSM!):

- [ ] Align Nodes in Line (shortcut in JOSM: L)
- [ ] Distribute Nodes (shortcut in JOSM: Shift-B)

Or does this exist somewhere/somehow in Vespucci (I don't think so)? It would a geat help in MANY situations to create a clean geometry or arranging multiple nodes. E.g. an entrance exactly in the middle of one side of a building (by selecting 3 nodes and using "Distribute Nodes").

I would think, that this should not be too complex to code (but I don't know).

**Comment by simonpoole** [comment]
Note the othogonalize/straighten function will straighten an unclosed line.

**Comment by jajajaneeneenee** [comment]
> Note the othogonalize/straighten function will straighten an unclosed line.

That's a good hint. I forgot that command and did not use it very often (or never?) I think, don't know why. I checked it now ...

It only seems to work if a line doesn't have too acute angles – indeed it only works if it has very few angles (or what is the limitation?) – perhaps that's why I didn't used this command very often, perhaps because I thought it would not work properly... I would prefer if it would work under every condition – it would be more straightforward and logic.

And unfortunately you cannot align a multi selection of **single nodes** in a line with this comand (e.g. 3 benches, 10 single trees on a straight road, ....) – a workaround would be to connect them with a line, use the command and extract the points again, but this is quite time-consuming ...

It also would be nice, if you could select some nodes of a closed way with more nodes and only straighten the selected ones (same as with single nodes). Then it would be a smart command ....

**Comment by simonpoole** [comment]
> 
> 
> > Note the othogonalize/straighten function will straighten an unclosed line.
> 
> That's a good hint. I forgot that command and did not use it very often (or never?) I think, don't know why. I checked it now ...
> 
> It only seems to work if a line doesn't have too acute angles – indeed it only works if it has very few angles (or what is the limitation?) – perhaps that's why I didn't used this command very often, perhaps because I thought it would not work properly... I would prefer if it would work under every condition – it would be more straightforward and logic.

It will not correct deviations from 90°/180° more than the "squaring threshold" in the advanced preferences, standard that is set to 15°.
> 
> And unfortunately you cannot align a multi selection of **single nodes** in a line with this comand (e.g. 3 benches, 10 single trees on a straight road, ....) – a workaround would be to connect them with a line, use the command and extract the points again, but this is quite time-consuming ...
> 
> It also would be nice, if you could select some nodes of a closed way with more nodes and only straighten the selected ones (same as with single nodes). Then it would be a smart command .... (and a second "distribute nodes" command would even be nicer ... or the premium command would be "distribute nodes with a (optional) entered distance" ... I am using such commands since a long time e.g,. in Adobe InDesign, very useful ...).

Please 1 feature request per issue (this one already violated that, but it should not be taken as licence to make the problem worse)..

**Comment by jajajaneeneenee** [comment]
> Please 1 feature request per issue (this one already violated that, but it should not be taken as licence to make the problem worse).

I apologize  ... for me, it was also connected to "Multi select actions" ... And I wanted to contribute my spontaneous  thoughts. It came from the ["align in circle" suggestion](https://github.com/MarcusWolschon/osmeditor4android/issues/881#issuecomment-475937281), so I thought I could add it here. Should I open a new issue now?

**Comment by simonpoole** [comment]
Yes it would be nice to have one issue per feature in particular because outside of multi-select they have nothing in common.

**Comment by simonpoole** [comment]
Most of 2. has been done in the improved relation editing support for 15.1

**Comment by simonpoole** [comment]
The new relation handling code works for multi-select too. The issue of doing the same op multiple time is tracked here https://github.com/MarcusWolschon/osmeditor4android/issues/1268
